app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/SendFiles.kt

```diff
@@ -0,0 +1,36 @@
+package org.kiwix.kiwixmobile.zim_manager.fileselect_view.effects
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import androidx.core.content.FileProvider
+import org.kiwix.kiwixmobile.BuildConfig
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskListItem.BookOnDisk
+import org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity
+
+class SendFiles(private val selectedBooks: List<BookOnDisk>) : SideEffect<Unit> {
+  override fun invokeWith(activity: Activity) {
+    val selectedFileShareIntent =
+      Intent(activity.applicationContext, LocalFileTransferActivity::class.java)
+
+    val selectedFileContentURIs = selectedBooks.mapNotNull {
+      if (Build.VERSION.SDK_INT >= 24) {
+        FileProvider.getUriForFile(
```

Comment:
an extension function could be good here if kept local and private to this file.

```fun BookOnDisk.toFileUri() =
if(Build.Version…) etc```

Reply:
[Commit](https://github.com/kiwix/kiwix-android/pull/1538/commits/a832e0f9bcbacf85cd20b0b32c4b71429b6cd7ab)
Could you have a look whether it is done right? I'm not 100% sure of what I'm doing in kt

Reply:
You made the extension on `List<BookOnDisk>` and not `BookOnDisk`

`fun BookOnDisk.toFileUri()` use this signature,

and I should see `selectedBooks.mapNotNull { it.toFileUri }`

The [kotlin docs](https://kotlinlang.org/docs/reference/extensions.html) are generally pretty concise as long as you don't start reading about coroutines